db/memtable.cc

```diff
@@ -756,10 +775,13 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
   Slice user_key = key.user_key();
   bool found_final_value = false;
   bool merge_in_progress = s->IsMergeInProgress();
-  bool const may_contain =
+  bool may_contain =
       nullptr == prefix_bloom_
           ? false
           : prefix_bloom_->MayContain(prefix_extractor_->Transform(user_key));
+  if (moptions_.memtable_whole_key_filtering) {
+    may_contain &= prefix_bloom_->MayContain(user_key);
```

Comment:
If there is whole key filtering. There is no reason to check prefix bloom at all.